### ISSUE TITLE
[4.0b7] Link to contact from tags view is generated to a 404 page

### ISSUE THREAD

**Issue by PhilETaylor** [joomla/joomla-cms#32248]
### Steps to reproduce the issue

Login to 4.0b7 and create a contact ensuring you give it a tag

Create a frontend menu item of type Tags -> Tagged Items and select the tag you gave the contact

Visit the frontend

Click the menu item to load the tagged list - note the contact is shown in a table

<img width="1182" alt="Screenshot 2021-02-02 at 20 46 24" src="https://user-images.githubusercontent.com/400092/106660310-bbca5a80-6597-11eb-9c66-527a722363d4.png">

**Click the contacts link - myName in this example**

### Expected result

Linked to the contact

### Actual result

<img width="861" alt="Screenshot 2021-02-02 at 20 46 52" src="https://user-images.githubusercontent.com/400092/106660358-cb49a380-6597-11eb-8e49-46d646b7fc6c.png">

404

Url is (in my case) 
http://127.0.0.1:4444/index.php/component/contact/contact/myname?catid=10&Itemid=101

**Comment by joomdonation** [comment]
@csthomas made a PR long time ago to fix this issue https://github.com/joomla/joomla-cms/pull/20979 . However, that PR is quite hard to follow.

**Comment by chmst** [comment]
I had the same issue and have tried with PR #20979. The patch resolves this problem, after patch no 404 and the contact is displayed.

**Comment by joomdonation** [comment]
Someone should ask @Hackwar to help with this issue. Look like we are having problem with **NomenuRules**. This is a hard issue and hard for someone not familiar with new routing code to fix, so better if @Hackwar could spend sometime to fix this issue.

**Comment by Hackwar** [comment]
I have this on my radar, unfortunately my time right now is limited. Hopefully will be able to get this fixed this month.

**Comment by Hackwar** [comment]
PR in #32695

**Comment by joomdonation** [comment]
Many thanks @Hackwar. Will test it.

**Comment by richard67** [comment]
Shall #20979 be closed, too, i.e. the new PR #32695 replaces that one?
